src/dbnode/storage/database.go

```diff
@@ -245,6 +247,19 @@ func (d *db) UpdateOwnedNamespaces(newNamespaces namespace.Map) error {
 		d.log.Error("failed to update schema registry", zap.Error(err))
 	}
 
+	// Always update the runtime options if they were set so that correct
+	// runtime options are set in the runtime options registry before namespaces
+	// are actually created.
+	for _, namespaceMetadata := range newNamespaces.Metadatas() {
+		id := namespaceMetadata.ID().String()
```

Comment:
Are we fine to alloc the string here?

Reply:
Yeah thankfully this only happens on changes.